docs/next_link_analysis.md

```diff
@@ -0,0 +1,108 @@
+- the Mean Reciprocal Rank (MRR) for the predictions is [0.43] on a holdout test set
+
+These results should not be regarded in terms of good or bad, rather they just establish a reference score. This serves as a benchmark to be beaten in future tailored predictions, which incorporate additional explanatory data regarding the user. This enables meaningful discussion and development of future models. These measures should be seen as the naiive or random prediction threshold. Any future models should be better than this measure.  
+
+## Data used
+
+The data used in the analysis is pulled from a single hour of usage (midday) of the GovUK website on 2020-11-13. This contains 844k unique records. As can be seen beneath, for a single `fullVisitorId`, there can be multiple values for `visitId`. It relates to hits which are classified as `PAGE`.
+
+## Illustration of data
+
+![visitor data](./images/visit_data_example.jpeg)
+
+As seen, pagePath represents the URL visited.
+
+## Rationale for considering only the preceding website
+
+Typically in time series processes, the most important data point is the current one, in determining the next data point. In a memoryless process, the only data point which is relevant is the current one and the following data point will be a random jump to one of the connected points in the space. In the context of the GovUK website, it is frequently the case that a visitor can only visit the next URL via the current URL.
+
+### Conditioning on time
+
+As we are focussing on the single hour of data, we have implicitly conditioned on time. Different  predictions would be made for different slices of time. This would mean that this approach could incoporate non-stationary effects and the impact of current events. As will be discussed beneath, the model could be run in real time, as it is effectively relying on aggregations and dictionary data access which is inherently fast.
+
+### Conditioning on location
+
+Location is not considered here, but would be possible given that there is information about geolocation of the user. This simply requires consideration of aggregation weights as a function of distance, discounting information from users further away.
+
+## Preparing the lagged data
+
+The data requires a lagged URL to be created in order to see the link between current and past data. We are trying to predict the URL to be visited, by referencing the lagged URL.
+
+Of note, a full 53% of the users on the government website only visit one link. This makes it impossible to create a lagged URL. These users data are removed from the dataset, as we cannot build a prediction conditioned on the past, as there is no history for the user.
+
+### Removing login, search and done activity
+
+Any URLs for search or login are removed. Search activity only accounts for [4%] of the remaining data, which when considered in combination with the number of users who only visit one URL, is indicative of external search engines.
```

Comment:
This could suggest user beliefs of how well Google Search performs vs. GOV.UK Search. Specifically, they search on Google because it returns better results than searching on GOV.UK. It's an avenue GOV.UK teams have wanted to explore further but have not had capacity to. I've written a Trello card to capture this as something we could revisit in the future.